launch/pipeline/service.py

```diff
@@ -59,9 +95,22 @@ class SequentialPipelineDescription(ServiceDescription):
     def __init__(
         self,
         items: List[SingleServiceDescription],
+        runtime: Runtime,
+        deployment: Deployment,
     ):
         super().__init__()
         self.items = items
+        self.runtime = runtime
+        self.deployment = deployment
+        self.service_name = f"pipeline{uuid.uuid4()}"[:MAX_SERVICE_NAME_SIZE]
+
+    def set_make_request(
```

Comment:
I'm a bit confused as to what this is useful for? Why mutate object state here -- it makes things more complicated. What use cases do you envision where mutation is strictly necessary?

Reply:
^ I updated the PR comment. We'll use it to call external service. If an object `foo` does not have `set_make_request` redefined, it will use the `call` method from the usercode. If `set_make_request` overrided `make_request`, that means we will use `make_request(req)` instead of `call(req)`. 

The trick is that on the backend side we can redefine `make_request` to use another service for example